xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/src/test/it/org/xwiki/flamingo/test/docker/VersionIT.java

```diff
@@ -645,4 +650,59 @@ void testDeleteVersionDontMessUpRightsWithCancellingEvent(TestUtils setup)
             historyPane.rollbackToVersion(latestVersionBeforeChanges);
         }
     }
+
+    @Test
+    @Order(10)
+    void getRevisionsWithCriteria(TestUtils testUtils, TestReference testReference,
+        LogCaptureConfiguration logCaptureConfiguration) throws Exception
+    {
+        testUtils.loginAsSuperAdmin();
+        testUtils.rest().delete(testReference);
+        testUtils.rest().savePage(testReference, "Some content", "Title");
+        testUtils.rest().savePage(testReference, "Some content 1", "Title");
+        testUtils.rest().savePage(testReference, "Some content 1 2", "Title");
+        testUtils.rest().savePage(testReference, "Some content 1 2 3", "Title");
+        testUtils.rest().savePage(testReference, "Some content 1 2 3 4", "Title");
+
+        ViewPage viewPage = testUtils.gotoPage(testReference);
+        HistoryPane historyPane = viewPage.openHistoryDocExtraPane();
+        assertEquals("5.1", historyPane.getCurrentVersion());
+        assertEquals(5, historyPane.getNumberOfVersions());
+
+        String currentTestReference = testUtils.serializeReference(testReference);
+        String targetTestReference = "xwiki:Test.getRevisionsWithCriteriaFoo.WebHome";
+        String script = String.format("""
+        {{velocity}}
+        #set ($myTest = "%s")
+        #set ($startAt = 0)
+        #set ($endAt = -1)
+        #set ($criteria = $xwiki.criteriaService.revisionCriteriaFactory.createRevisionCriteria('', $minorVersions))
+        #set ($range = $xwiki.criteriaService.rangeFactory.createRange($startAt, $endAt))
+        #set ($discard = $criteria.setRange($range))
+        #set ($myDoc = $xwiki.getDocument($myTest))
+        #set ($xwikiDoc = $myDoc.document)
+        #set ($discard = $myDoc.document.loadArchive($xcontext.context))
+        XWiki Doc: $xwikiDoc
+        #set ($revisions = $xwikiDoc.getRevisions($criteria, $xcontext.context))
+        Revision: $revisions
+        #set ($newRef = $services.model.resolveDocument("%s"))
+        #set ($discard = $xwikiDoc.setDocumentReference($newRef))
+        XWiki Doc: $xwikiDoc
+        #set ($revisions = $xwikiDoc.getRevisions($criteria, $xcontext.context))
+        Revision: $revisions
+        {{/velocity}}
+        """, currentTestReference, targetTestReference);
+
+        DocumentReference scriptReference = new DocumentReference("Script", testReference.getLastSpaceReference());
```

Comment:
You can replace this plumbing by the use of TestUtils#executeWiki (it executes the passed wiki content and return back the result).

Reply:
hmm I will look that thanks

Reply:
Note that it also means adding `test.prchecker.excludePattern=.*:Test\\.Execute\\..*` to the @UITest. We might decide one day to make it the default, I guess (was too lazy to start this conversation back then...).

Reply:
So I did use `executeWikiPlain` but it's not yet available in 15.10.x so I'll need to move it there too.